src/gtk/textctrl.cpp

```diff
@@ -1525,6 +1559,12 @@ void wxTextCtrl::GTKOnTextChanged()
 {
     if ( IgnoreTextUpdate() )
         return;
+    if( IsMultiLine() )
+    {
+        auto count = gtk_text_buffer_get_char_count( m_buffer );
+        if( m_maxlen > 0 && count >= m_maxlen )
```

Comment:
Why is `>=` used here? Shouldn't it be `>`?

```suggestion
        if ( m_maxlen > 0 && count > m_maxlen )
```

Reply:
@vadz , Changing this will bring back bad event generation behavior.

Reply:
I have no idea what is this supposed to mean, but it doesn't seem obvious at all why we should ignore the changes when the char count is exactly equal to `m_maxlen`, so you need to either explain it (in a comment) if it's somehow correct or fix the code if it isn't.

Reply:
@vadz ,
I tried to explain it in the comment. Please let me know if you need a better wording...
Thx.

Reply:
This doesn't explain anything. When the count is equal to `m_maxlen`, there should be no events about overflowing it at all, so if anything, this indicates that there is a bug somewhere.

Reply:
This seems to happens because after typing the character that exceeds the max, it is deleted in `au_insert_text_callback`.
But this delete action generates another gtk text event that calls this function.

I think the correct fix is to add this after `gtk_text_buffer_delete`:
```c++
win->IgnoreNextTextUpdate();
```

With this change, it seems all this code in `GTKOnTextChanged` is not needed anymore.
(But please test)

Reply:
@MaartenBent ,
Made the changes and they seemed to work.

Could you try them please if you have a spare minute, please?

Reply:
I tried the text example in Ubuntu gtk3 and there it works when typing and pasting.

Reply:
@MaartenBent ,
Thx for testing.

Hopefully it is now good.